packages/task/src/browser/task-service.ts

```diff
@@ -129,24 +144,27 @@ export class TaskService implements TaskConfigurationClient {
             }
 
             const taskConfiguration = event.config;
-            const taskIdentifier =
-                taskConfiguration
-                    ? ContributedTaskConfiguration.is(taskConfiguration)
-                        ? `${taskConfiguration._source}: ${taskConfiguration.label}`
-                        : `${taskConfiguration.type}: ${taskConfiguration.label}`
-                    : `${event.taskId}`;
-
-            if (event.code !== undefined) {
-                const message = `Task ${taskIdentifier} has exited with code ${event.code}.`;
-                if (event.code === 0) {
-                    this.messageService.info(message);
+            const provider = this.taskProviderRegistry.getProvider(taskConfiguration.source);
+            if (!provider || !provider.attach) {
+                const taskIdentifier =
+                    taskConfiguration
+                        ? ContributedTaskConfiguration.is(taskConfiguration)
+                            ? `${taskConfiguration._source}: ${taskConfiguration.label}`
+                            : `${taskConfiguration.type}: ${taskConfiguration.label}`
+                        : `${event.taskId}`;
+
+                if (event.code !== undefined) {
+                    const message = `Task ${taskIdentifier} has exited with code ${event.code}.`;
+                    if (event.code === 0) {
+                        this.messageService.info(message);
```

Comment:
The `message` is only ever used once, is it necessary to add a variable?

Reply:
That's existing code, just moved inside the 'if' statement.  I have noted though that it does seem to be the Typescript style to inline everything into one big statement, rather than breaking it down into smaller, more readable statements and naming intermediate values.  Some examples in the style guide might be useful.  I'm happy to change this if this is the style you all prefer.